### PR TITLE
feat: added dup check for image and asset-name and white space

### PR DESCRIPTION
**Pull request by mrabdibdi-anvil**

### CODE REVIEW

File: core/src/validator/src/rules/key-white-space.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { metadataValidator } from "../utils/metadataChecks.ts";
+import { findWhitespace } from "../utils/whiteSpace.ts";
+
+export class KeyWhiteSpace extends BaseValidator {
+  constructor() {
+    const id = "key-white-space";
+    super(id);
+  }
+
+  async Execute(
```
Comment (pChausseC) [inline core/src/validator/src/rules/key-white-space.ts]: this.Logic is synchronous so Execute should be synchronous and shouldnt return a promise

File: core/src/validator/src/rules/duplicate-asset-name.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { metadataValidator } from "../utils/metadataChecks.ts";
+
+export class DuplicateAssetName extends BaseValidator {
+  constructor() {
+    const id = "duplicate-asset-name";
+    super(id);
+  }
+
+  async Execute(
+    asset_name: string,
```
Comment (pChausseC) [inline core/src/validator/src/rules/duplicate-asset-name.ts]: normalize variable name casing (here snake_case is used while camelCase is used everywhere else)

File: core/src/validator/__tests__/test.white-space.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  const validatorsReceivedFromFrontend: [keyof typeof mapping] = [
+    "KeyWhiteSpace",
+  ];
+
+  const mainValidator = new Decorator("Main");
+  for (const validator of validatorsReceivedFromFrontend) {
+    mainValidator.Enable(new mapping[validator]());
+  }
+
+  for (const asset_metadata of metadata) {
```
Comment (pChausseC) [inline core/src/validator/__tests__/test.white-space.ts:36]: normalize variable name casing (here snake_case is used while camelCase is used everywhere else)

**Review by pChausseC — CHANGES_REQUESTED** [review]

File: core/src/validator/src/rules/key-white-space.ts (hunk 1 above)
Comment (mrabdibdi-anvil) [reply core/src/validator/src/rules/key-white-space.ts]: removed

File: core/src/validator/__tests__/test.white-space.ts (hunk 3 above)
Comment (mrabdibdi-anvil) [reply core/src/validator/__tests__/test.white-space.ts:36]: normalized to camel case everywhere

File: client/package.json [hunk 4]
```diff
@@ -24,7 +24,6 @@
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
     "@ada-anvil/metadraft-validator": "file:../core/src/validator/npm",
-    "react-dropzone": "^14.2.3",
     "tailwind-merge": "^2.4.0",
```
Comment (pChausseC) [inline client/package.json]: I believe this package is needed

File: core/src/validator/__tests__/test.case.ts [hunk 5]
```diff
@@ -175,33 +164,36 @@ Deno.test("KeyLowerCase - withWarning", async () => {
...
-      asset_metadata,
-      metadata,
-    );
+    mainValidator.Execute(asset_metadata.assetName, asset_metadata, metadata);
   }
 
   const result = mainValidator.GetResults();
 
-  console.debug(JSON.stringify(result));
+  // console.debug(JSON.stringify(result));
```
Comment (pChausseC) [inline core/src/validator/__tests__/test.case.ts]: remove commented clog

**Review by pChausseC — CHANGES_REQUESTED** [review]

File: core/src/validator/__tests__/test.case.ts (hunk 5 above)
Comment (mrabdibdi-anvil) [reply core/src/validator/__tests__/test.case.ts]: Done, when it will be merged, I will also add a global logger to remove the remaining Clogs

**Review by pChausseC — APPROVED** [review]

**Pull request merged**